starky/src/get_challenges.rs

```diff
@@ -33,21 +30,16 @@ fn get_challenges<F, C, S, const D: usize>(
 where
     F: RichField + Extendable<D>,
     C: GenericConfig<D, F = F>,
-    S: Stark<F, D>,
 {
     let num_challenges = config.num_challenges;
 
     let mut challenger = Challenger::<F, C::Hasher>::new();
 
     challenger.observe_cap(trace_cap);
 
-    let permutation_challenge_sets = permutation_zs_cap.map(|permutation_zs_cap| {
-        let tmp = get_n_permutation_challenge_sets(
-            &mut challenger,
-            num_challenges,
-            stark.permutation_batch_size(),
-        );
-        challenger.observe_cap(permutation_zs_cap);
+    let lookup_challenge_set = auxiliary_polys_cap.map(|auxiliary_polys_cap| {
+        let tmp = get_grand_product_challenge_set(&mut challenger, num_challenges);
+        challenger.observe_cap(auxiliary_polys_cap);
         tmp
```

Comment:
Can you use something like `inspect` here to get rid of the `tmp` var? Perhaps not, given it's not an iterator. Still. Hmm.